core/src/test/java/io/confluent/rest/SslTest.java

```diff
@@ -226,6 +227,38 @@ public void testHttpsOnly() throws Exception {
     }
   }
 
+  @Test
+  public void test_WhenCalledWithRestrictedCipher_ConnFails() throws Exception {
+    TestMetricsReporter.reset();
+    Properties props = new Properties();
+    String uri = "https://localhost:8080";
+    props.put(RestConfig.LISTENERS_CONFIG, uri);
+    props.put(RestConfig.METRICS_REPORTER_CLASSES_CONFIG, "io.confluent.rest.TestMetricsReporter");
+    configServerKeystore(props);
+    // Restrict HTTPs Server to cipher TLS_AES_128_GCM_SHA256
+    props.put(RestConfig.SSL_CIPHER_SUITES_CONFIG, "TLS_AES_128_GCM_SHA256");
+    TestRestConfig config = new TestRestConfig(props);
+    SslTestApplication app = new SslTestApplication(config);
+    try {
+      app.start();
+
+      // Correct Cipher, returns 200.
+      // TLSv1.3 should be specified with TLS_AES_128_GCM_SHA256 only for test-setup, as this cipher is specific to TLSv1.3.
+      // NOTE - the behaviour of restricting cipher in REST is agnostic of TLS protocols.
```

Comment:
On this point, there are ciphers suite that only compatible with certain TLS version, for example if we restrict to cipher suite `TLS_AES_128_GCM_SHA256` then only TLS1.3 is supported on the server.